### PR TITLE
Change of insertion order to better reflect the method's internal logic

### PR DESCRIPTION
**Pull request by gonsoos**
Homework Number: W09H01

A short description of the changes proposed in the pull request:
Currently, `iterator()` makes it seem, as if `integerStringTupleHashSet0.insert(tuple0Duplicate)` discarded the tuple in favor of the already existing `tuple0`.
The catch is that TupleHashSet's insert method replaces the existing tuple with an equal tuple as opposed to TuplePool's insert method, which discards a given equal tuple.
That's why I changed the order of insertion to better reflect the method's internal logic.

You could also do the following instead, as TupleHashSet's `contains` method makes use of Tuple's `equals` method.
```java
- assertTrue(tuples.contains(tuple0));
+ assertTrue(tuples.contains(tuple0Duplicate));
```
:notes: _In the end, it doesn't even matter_ :notes: This "issue" may be kind of nitpicky, so do as you please :)

[//]: <> (change [ ] to [x] in order to check the box)
- [x] Solution is not getting revealed
- [x] Files are formatted

### CODE REVIEW

File: W09H01/src/test/pgdp/hashset/TupleHashSetTest.java [hunk 1]
```diff
@@ -415,14 +415,5 @@ void iteratorFullHashSet() {
...
     }
     assertEquals(353, tuplesInHashSet.size());
-    Arrays.stream(tuples).forEach(integerIntegerTupleHashSet::insert);
-    
-    tuplesInHashSet = new ArrayList<>();
-    for (Tuple<Integer, Integer> integerIntegerTuple : integerIntegerTupleHashSet) {
-      tuplesInHashSet.add(integerIntegerTuple);
-    }
-    assertEquals(353, tuplesInHashSet.size());
-
```
Comment (gonsoos) [inline W09H01/src/test/pgdp/hashset/TupleHashSetTest.java:425]: This is simply code duplication, right?

**Comment by LadnerJonas** [comment]
> Currently, iterator() makes it seem, as if integerStringTupleHashSet0.insert(tuple0Duplicate) discarded the tuple in favor of the already existing tuple0.

Instruction:
>insert(): Mit dieser Methode soll ein Tupel in den Pool eingefügt werden. Falls ein gleiches Tupel bereits vorhanden ist, wird das übergebene Tupel verworfen. Es soll das Tupel zurückgegeben werden, welches letztendlich im Pool gespeichert ist.

Sorry, but i cannot quite understand why this change should help test the iterator.

File: W09H01/src/test/pgdp/hashset/TupleHashSetTest.java (hunk 1 above)
Comment (LadnerJonas) [reply W09H01/src/test/pgdp/hashset/TupleHashSetTest.java:425]: Yeah it is duplicate code, but as the HashSet is not cleared in between the "instert-blocks", this should check, if the iterator does not change the saved data.

Comment (gonsoos) [reply W09H01/src/test/pgdp/hashset/TupleHashSetTest.java:425]: Ah, right! I did not see that! I'll close this pull request :+1:

**Pull request closed without merging**